src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -92,7 +92,7 @@ public static partial class RequestDelegateFactory
     private static readonly MemberExpression FormFilesExpr = Expression.Property(FormExpr, typeof(IFormCollection).GetProperty(nameof(IFormCollection.Files))!);
     private static readonly MemberExpression StatusCodeExpr = Expression.Property(HttpResponseExpr, typeof(HttpResponse).GetProperty(nameof(HttpResponse.StatusCode))!);
     private static readonly MemberExpression CompletedTaskExpr = Expression.Property(null, (PropertyInfo)GetMemberInfo<Func<Task>>(() => Task.CompletedTask));
-    private static readonly NewExpression CompletedValueTaskExpr = Expression.New(typeof(ValueTask<object>).GetConstructor(new[] { typeof(Task) })!, CompletedTaskExpr);
+    private static readonly ConstantExpression EmptyResultValueTaskExpr = Expression.Constant(new ValueTask<object?>(EmptyHttpResult.Instance));
```

Comment:
To avoid adding to https://github.com/dotnet/aspnetcore/issues/41435, can we use an alternative syntax for this? You can check out [the diff in this PR](https://github.com/dotnet/aspnetcore/pull/43380/files#diff-f38f5cec66cf8010067587a59799e7aeb620388d108d0c58e63aa23849eb6eebR102-R104) for an example.

Reply:
We can definitely replace it with the new expression instead. I noticed you included one of the changes already in your PR that uses the same expression so feel free to include the change for this issue in your PR instead to prevent conflicts, otherwise the change is now done here.